app/src/userpages/components/StreamPage/Show/index.jsx

```diff
@@ -286,29 +289,39 @@ function StreamLoader(props: Props) {
     const propsRef = useRef(props)
     propsRef.current = props
     const { id: streamId } = props.match.params || {}
+    const fail = useFailure()
 
     const initStream = useCallback(async (id: StreamId) => {
         let { current: currentProps } = propsRef
         await propsRef.current.getKeys()
         if (!isMounted()) { return }
         currentProps = propsRef.current
-        return Promise.all([
-            currentProps.getStream(id).then(async () => {
-                if (!isMounted()) { return }
-                // get stream status before copying state to edit stream object
-                try {
-                    // the status query might fail due to cassandra problems.
-                    // Ignore error to prevent the stream page from getting stuck while loading
-                    await currentProps.refreshStreamStatus(id)
-                } catch (e) {
-                    console.warn(e)
-                }
-                currentProps.openStream(id)
-                currentProps.initEditStream()
-            }),
-            currentProps.getMyStreamPermissions(id),
-        ])
-    }, [isMounted, propsRef])
+
+        try {
+            await Promise.all([
+                currentProps.getStream(id).then(async () => {
+                    if (!isMounted()) { return }
+                    // get stream status before copying state to edit stream object
+                    try {
+                        // the status query might fail due to cassandra problems.
+                        // Ignore error to prevent the stream page from getting stuck while loading
+                        await currentProps.refreshStreamStatus(id)
+                    } catch (e) {
+                        console.warn(e)
+                    }
+                    currentProps.openStream(id)
+                    currentProps.initEditStream()
+                }),
+                currentProps.getMyStreamPermissions(id),
+            ]).catch(handleLoadError)
```

Comment:
The other change to the old logic here is this `.catch`. This, and of course the following `catch (e)` block.